CRM/Core/ManagedEntities.php

```diff
@@ -600,16 +479,17 @@ protected function isModuleDisabled(string $module): bool {
 
   /**
    * @param array $declarations
-   *
-   * @return array
+   * @param string $moduleName
+   *   Filter to a single module.
    */
-  protected function cleanDeclarations(array $declarations): array {
-    foreach ($declarations as $name => &$declare) {
-      if (!array_key_exists('name', $declare)) {
-        $declare['name'] = $name;
+  protected function setDeclarations(array $declarations, $moduleName = NULL) {
+    $this->declarations = [];
```

Comment:
@totten I have been pushing in the direction of ephemeral. A few versions ago I hit some bugs caused by them hanging around from one reconcile to the next and they were solved by the addition of this line.
I would be happy to push it further in that direction and make them local variables passed between functions instead of class variables.

Reply:
https://github.com/civicrm/civicrm-core/pull/23251